### PR TITLE
[FLINK-11278] [docs] Add documentation for TableAPI&SQL in scala-shell

### PR DESCRIPTION
**Pull request by WeiZhong94**
## What is the purpose of the change

This pull request adds documentation for TableAPI&SQL in scala-shell.

## Brief change log

  - add documentation for TableAPI&SQL in scala-shell

## Verifying this change
This change is not need to add verify  

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (docs)

### CODE REVIEW

File: docs/ops/scala_shell.md [hunk 1]
```diff
@@ -36,9 +36,10 @@ cluster, please see the Setup section below.
 
 ## Usage
 
-The shell supports Batch and Streaming.
-Two different ExecutionEnvironments are automatically prebound after startup.
-Use "benv" and "senv" to access the Batch and Streaming environment respectively.
+The shell supports Batch, Streaming, Table API and SQL. 
+Four different Environments are automatically prebound after startup. 
+Use “benv” and “senv” to access the Batch and Streaming ExecutionEnvironment respectively.
```
Comment (hequn8128) [inline docs/ops/scala_shell.md]: Wrong double quotation marks for benv and senv

File: docs/ops/scala_shell.md [hunk 2]
```diff
@@ -36,9 +36,10 @@ cluster, please see the Setup section below.
 
 ## Usage
 
-The shell supports Batch and Streaming.
-Two different ExecutionEnvironments are automatically prebound after startup.
-Use "benv" and "senv" to access the Batch and Streaming environment respectively.
+The shell supports Batch, Streaming, Table API and SQL. 
+Four different Environments are automatically prebound after startup. 
+Use “benv” and “senv” to access the Batch and Streaming ExecutionEnvironment respectively. 
+Use "btenv" and “stenv” to access BatchTableEnvironment and StreamTableEnvironment respectively.
```
Comment (hequn8128) [inline docs/ops/scala_shell.md]: Wrong double quotation marks for stenv

File: docs/ops/scala_shell.md [hunk 3]
```diff
@@ -85,6 +86,113 @@ Note, that in the Streaming case, the print operation does not trigger execution
...
+  }
+Scala-Flink> val split = new $Split
+Scala-Flink> textSource.join(split('text) as 'word).
+    groupBy('word).select('word, 'word.count as 'count).
+    toDataSet[(String, Long)].print
+{% endhighlight %}
+</div>
+</div>
+
+Note, that the $ prefix of the TableFunction classname is a walkaround of a scala issue about incorrectly generated inner class name.
```
Comment (hequn8128) [inline docs/ops/scala_shell.md]: walkaround => workaround
classname => class name

**Comment by WeiZhong94** [comment]
@hequn8128 Thanks for your review! I have updated the PR according to your comments.

File: docs/ops/scala_shell.md (hunk 3 above)
Comment (sunjincheng121) [reply docs/ops/scala_shell.md]: How about  `Note, that using $  as a prefix for the class name of TableFunction is a workaround of the issue that scala incorrectly generated inner class name.` ?

Comment (WeiZhong94) [reply docs/ops/scala_shell.md]: That sounds better. I have rewritten the sentence.

**Comment by sunjincheng121** [comment]
Thanks @WeiZhong94 and @hequn8128 !
LGTM. +1 to merged.

Bests,
Jincheng

**Comment by sunjincheng121** [comment]
Merging...

**Pull request closed without merging**